### PR TITLE
p7zip: modernize, fix audit warnings

### PR DESCRIPTION
**Pull request by saketkc**

### CODE REVIEW

**Comment by saketkc** [comment]
@BrewTestBot test this please.

**Comment by saketkc** [comment]
Good to merge @mikemcquaid?

File: Library/Formula/p7zip.rb [hunk 1]
```diff
@@ -11,23 +9,23 @@ class P7zip < Formula
...
     if Hardware.is_32_bit? or build.build_32_bit?
-      mv 'makefile.macosx_32bits', 'makefile.machine'
+      mv "makefile.macosx_32bits", "makefile.machine"
     else
-      mv 'makefile.macosx_64bits', 'makefile.machine'
+      mv "makefile.macosx_64bits", "makefile.machine"
     end
 
+    # install.sh chmods to 444, which is bad and breaks uninstalling
+    inreplace "install.sh", /chmod (444|555).*/, ""
```
Comment (mikemcquaid) [inline Library/Formula/p7zip.rb]: Can we submit an issue upstream about this?

Comment (mikemcquaid) [inline Library/Formula/p7zip.rb]: Oh, any reason to move this from the `chmod` below?

**Comment by mikemcquaid** [comment]
FYI: generally we don't want PRs that are only fixing audits/modernising unless they are also adding a test or fixing a new issue.

**Comment by saketkc** [comment]
It fails on `linuxbrew`. I originally submitted a patch there, but that could not be merged due to `travis` complaining about the audit warning.
https://github.com/Homebrew/linuxbrew/pull/231#discussion_r22686207

**Comment by mikemcquaid** [comment]
Ok. Would like us to submit this upstream, though.

**Comment by saketkc** [comment]
Submitted: http://sourceforge.net/p/sevenzip/patches/253/

**Comment by saketkc** [comment]
Should I close this, or would it make sense to merge?

**Comment by mikemcquaid** [comment]
Merged this time, thanks!

**Pull request closed without merging**